pkg/util/sys/device.go

```diff
@@ -399,35 +411,106 @@ func parseUdevInfo(output string) map[string]string {
+	if string(sig[:]) == "bluestore block device" {
+		return false, "has BlueStore device label", nil
+	}
+
+	// available
+	return true, "", nil
 }
 
-func inventoryDevice(executor exec.Executor, devicePath string) (CephVolumeInventory, error) {
-	var CVInventory CephVolumeInventory
+// if this function returns an error, the disk should not be used (should be considered unavailable)
+func isDeviceAvailable(executor exec.Executor, devicePath string) (available bool, rejectedReasons string, outErr error) {
+	output, err := executor.ExecuteCommandWithOutput("lsblk", devicePath,
+		"--bytes", "--nodeps", "--noheadings", "--output", "RM,RO,SIZE,TYPE")
+	if err != nil {
+		return false, "", errors.Wrapf(err, "failed to determine if device %q is available", devicePath)
+	}
+	items := strings.Fields(output)
+	if len(items) != 4 {
+		return false, "", errors.Errorf(
+			"failed to determine if device %q is available. lsblk output should have 4 fields but has %d: %s", devicePath, len(items), items)
+	}
+	rm := items[0]
+	ro := items[1]
+	sizeStr := items[2]
+	devType := items[3]
+
+	rejectedList := []string{}
+
+	// must not be removable
+	if rm != "0" { // 0 means not removable, 1 means removable
+		rejectedList = append(rejectedList, "removable")
+	}
 
-	args := []string{"inventory", "--format", "json", devicePath}
-	inventory, err := executor.ExecuteCommandWithOutput("ceph-volume", args...)
+	// must not be read-only
+	if ro != "0" { // 0 means not read-only, 1 means read-only
+		rejectedList = append(rejectedList, "read-only")
+	}
+
+	// must be large enough
+	sizeBytes, err := strconv.Atoi(sizeStr)
 	if err != nil {
-		return CVInventory, fmt.Errorf("failed to execute ceph-volume inventory on disk %q. %v", devicePath, err)
+		return false, "", errors.Wrapf(err,
+			"failed to determine if device %q is available. failed to interpret reported size %q", devicePath, sizeStr)
+	}
+	if sizeBytes < 5368709120 /* 5368709120 = 5GB */ {
+		rejectedList = append(rejectedList, "insufficient space (< 5GB)")
 	}
 
-	bInventory := []byte(inventory)
-	err = json.Unmarshal(bInventory, &CVInventory)
+	// must be raw device or partition
+	// rejects devType == "lvm" if is an LVM device because LVM is handled in other functions
+	if devType != DiskType && devType != PartType {
+		rejectedList = append(rejectedList,
+			fmt.Sprintf(`device type %q is not acceptable; should be raw device ("disk") or partition ("part")`, devType))
+	}
+
+	available, rejectedReason, err := rejectDeviceIfBluestore(devicePath)
 	if err != nil {
-		return CVInventory, fmt.Errorf("error unmarshalling json data coming from ceph-volume inventory %q. %v", devicePath, err)
+		return false, "", err
+	}
+	if !available {
+		rejectedList = append(rejectedList, rejectedReason)
 	}
 
-	return CVInventory, nil
+	// do not need to check if disk is a ceph-disk member because Rook has never used ceph-disk
+	// disks, is not intended to adopt those disks, and is not in danger of stepping on those
```

Comment:
@leseb @travisn just wanting to make sure you see this and don't think this is wrong. Checking for ceph-disk is really challenging.

Reply:
LGTM